internal/controller/bucket/conversions.go

```diff
@@ -117,40 +117,59 @@ func LateInitialize(params *v1alpha1.BucketParameters, obs *domain.Bucket) bool
 	params.RP = li.LateInitializeStringPtr(params.RP, obs.Rp)
 	if params.SchemaType == "" && obs.SchemaType != nil && string(*obs.SchemaType) != "" {
 		params.SchemaType = string(*obs.SchemaType)
-		return true
+		li.SetChanged()
 	}
-	sort.SliceStable(params.RetentionRules, func(i, j int) bool {
-		return params.RetentionRules[i].Type < params.RetentionRules[j].Type
-	})
-	sort.SliceStable(obs.RetentionRules, func(i, j int) bool {
-		return string(obs.RetentionRules[i].Type) < string(obs.RetentionRules[j].Type)
-	})
-	for i := range params.RetentionRules {
-		if params.RetentionRules[i].Type == string(obs.RetentionRules[i].Type) {
-			params.RetentionRules[i].ShardGroupDurationSeconds = li.LateInitializeInt64Ptr(params.RetentionRules[i].ShardGroupDurationSeconds, obs.RetentionRules[i].ShardGroupDurationSeconds)
+
+	if len(params.RetentionRules) == 0 && len(obs.RetentionRules) > 0 {
+		sort.SliceStable(obs.RetentionRules, func(i, j int) bool {
+			return string(obs.RetentionRules[i].Type) < string(obs.RetentionRules[j].Type)
+		})
+
+		rr := make([]v1alpha1.RetentionRule, len(obs.RetentionRules))
+		for i := range obs.RetentionRules {
+			rr = append(rr, v1alpha1.RetentionRule{
```

Comment:
When you initialize with `make`, you need to assign by index. Otherwise it appends [after](https://go.dev/play/p/gHImW78UUP_e) the empty places it initialized.

Reply:
Addressed in 5e7a6bd.